File: part1/hello/hello_client.cpp
// Apache Thrift Hello World C++ Client

#include "gen-cpp/HelloSvc.h"
#include <thrift/transport/TSocket.h>
#include <thrift/transport/TBufferTransports.h>
#include <thrift/protocol/TBinaryProtocol.h>
// Use boost for older versions of thrift
//#include <boost/make_shared.hpp>
#include <iostream>
#include <memory>
#include <string>

using namespace apache::thrift::transport;  
using namespace apache::thrift::protocol;   
// Use boost for older versions of thrift
//using boost::make_shared;
using std::make_shared;

int main() {
    auto trans_ep = make_shared<TSocket>("localhost", 9090);
    auto trans_buf = make_shared<TBufferedTransport>(trans_ep);
    auto proto = make_shared<TBinaryProtocol>(trans_buf);
    HelloSvcClient client(proto);

    trans_ep->open();
    std::string msg;
    client.hello_func(msg);
    std::cout << "[Client] received: " << msg << std::endl;
    trans_ep->close();
}

File: part3/script/php/hello_client.php
<?php
  require 'vendor/autoload.php';
  require 'gen-php/helloSvc.php';

  $msg = 'No server response';

  if ($_SERVER['REQUEST_METHOD'] === 'POST' && 
      array_key_exists('hello_form', $_POST) && 
      $_POST['hello_form'] == "Submit") {
    try {
      $trasn_ep = new Thrift\Transport\TSocket('localhost', 9095);
      $trans_buf = new Thrift\Transport\TBufferedTransport($trasn_ep);
      $proto = new Thrift\Protocol\TBinaryProtocol($trans_buf);
      $client = new helloSvcClient($proto);

      $trasn_ep->open();
      $msg = $client->getMessage($_POST['user_name']);
      $trasn_ep->close();
    } catch (Thrift\Exception\TException $tx) {
      error_log( 'TException: '.$tx->getMessage()."\n" );
    }
  }
?>

<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <title>Hello Thrift</title>
  </head>
  <body>
    <form name="hello_form" method="post" action="hello_client.php">
      Name: 
      <input type="text" name="user_name">
      <input type="submit" name="hello_form" value="Submit">
    </form>
    <br>
    <div><?php echo "$msg"; ?></div>
  </body>
</html>

